_posts/2019-09-02-feature-less.md

```diff
@@ -0,0 +1,20 @@
+---
+layout: post
+title: ???
+---
+
+## Slimmed gfx-rs
+
+We've done a few big steps towards making gfx-hal easier to build and maintain. First of all, we removed the "magical" dependencies, such as `failure` and `derivative` in favor of straight and simple code. That sped up the gfx-hal build by a factor of 8.5X.
+
+Secondly, the "typed" layer of gfx-hal got removed. Previously, we recognized that the view of the API for the backends (when implementing it) needs to be different from the users (to use it). Backends just need to implement "raw" interfaces, while users need more type safety, such as limiting the set of operations available on a command buffer when recording a render pass. This duality of APIs in gfx-hal worked to an extent. Providing safety was impossible without imposing some restrictions or overhead. At the end of the day, we decided that the "typed" (user-facing) layer is still useful, but it doesn't have to be a part of gfx-hal. So we removed it, recommending `rendy-command` as a replacement. This slimmed up gfx-hal API surface and allowed us to straighten up the terminology (no more "RawXXX" or "XxxTyped").
```

Comment:
Can we elaborate on the restrictions and/or overhead caused by providing the typed layer?

Reply:
From what I recall in gfx-hal-0.1 it was mostly type level stuff, so it mostly compiled out, but there were definite drawbacks in terms of it making the rustdoc much much harder to follow.